### ISSUE TITLE
ibm_dns_custom_resolver can not be created with two locations on the same subnet

### ISSUE THREAD

**Issue by powellquiring** [IBM-Cloud/terraform-provider-ibm#3746]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform IBM Provider Version

```
bug-dns-custom-resolver $ terraform -v
Terraform v1.1.9
on darwin_amd64
+ provider registry.terraform.io/ibm-cloud/ibm v1.40.1
```
### Affected Resource(s)

* ibm_dns_custom_resolver

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "ibm_resource_instance" "dns" {
  name              = var.prefix
  resource_group_id = local.resource_group
  location          = "global"
  service           = "dns-svcs"
  plan              = "standard-dns"
}

resource "ibm_dns_custom_resolver" "location" {
  name        = var.prefix
  instance_id = ibm_resource_instance.dns.guid
  locations {
    subnet_crn = ibm_is_subnet.front.crn
    enabled    = true
  }
  locations {
    subnet_crn = ibm_is_subnet.front.crn
    // fix by using back instead of front
    // subnet_crn = ibm_is_subnet.back.crn
    enabled = true
  }
}
```
apply:
```
ibm_dns_custom_resolver.location: Creating...
╷
│ Error: To meet high availability status, configure custom resolvers with a minimum of two resolver locations. A maximum of four locations can be configured within the same subnet location.
│
│   with ibm_dns_custom_resolver.location,
│   on dns.tf line 9, in resource "ibm_dns_custom_resolver" "location":
│    9: resource "ibm_dns_custom_resolver" "location" {
│
```

As the comment indicates if a different subnet is used this issue can be worked around.

You can find the full simple example: https://github.com/powellquiring/tfbugs/tree/master/bug-dns-custom-resolver

**Comment by MalarvizhiK** [comment]
Hi @powellquiring Add two locations with 2 different subnets for a customer resolver (CR). Minimum of 2 locations is needed to support High availability in CR.

**Comment by MalarvizhiK** [comment]
I have sent my terraform code to @powellquiring . I am unable to recreate the issue in Production. With high availability enabled, I am able to add 2 locations with same subnet.

**Comment by powellquiring** [comment]
```
cd /tmp
git clone https://github.com/powellquiring/tfbugs
cd bug-dns-custom-resolver
terraform init
terraform apply
...
Plan: 6 to add, 0 to change, 0 to destroy.
ibm_is_vpc.location: Creating...
ibm_resource_instance.dns: Creating...
ibm_is_vpc.location: Still creating... [10s elapsed]
ibm_resource_instance.dns: Still creating... [10s elapsed]
ibm_resource_instance.dns: Creation complete after 15s [id=crn:v1:bluemix:public:dns-svcs:global:a/713c783d9a507a53135fe6793c37cc74:1352fad7-b477-4874-8b6d-0f28ef32a7c7::]
ibm_is_vpc.location: Creation complete after 15s [id=r006-a3eae92a-0eae-4cbc-a4ea-2a2907435a30]
ibm_is_vpc_address_prefix.locations: Creating...
ibm_is_vpc_address_prefix.locations: Creation complete after 2s [id=r006-a3eae92a-0eae-4cbc-a4ea-2a2907435a30/r006-987a045e-83ce-42f6-bd34-1dbfc36fafa1]
ibm_is_subnet.back: Creating...
ibm_is_subnet.front: Creating...
ibm_is_subnet.back: Still creating... [10s elapsed]
ibm_is_subnet.front: Still creating... [10s elapsed]
ibm_is_subnet.back: Creation complete after 13s [id=0717-b8ddfd42-f333-4454-9598-3984b22326ce]
ibm_is_subnet.front: Still creating... [20s elapsed]
ibm_is_subnet.front: Creation complete after 25s [id=0717-403b20b0-a7ee-41aa-b2b3-6f4aeb488a1f]
ibm_dns_custom_resolver.location: Creating...
╷
│ Error: To meet high availability status, configure custom resolvers with a minimum of two resolver locations. A maximum of four locations can be configured within the same subnet location.
│
│   with ibm_dns_custom_resolver.location,
│   on dns.tf line 9, in resource "ibm_dns_custom_resolver" "location":
│    9: resource "ibm_dns_custom_resolver" "location" {
│
╵
```
Here is the example:
```
cat dns.tf
resource "ibm_resource_instance" "dns" {
  name              = var.prefix
  resource_group_id = local.resource_group
  location          = "global"
  service           = "dns-svcs"
  plan              = "standard-dns"
}

resource "ibm_dns_custom_resolver" "location" {
  name        = var.prefix
  instance_id = ibm_resource_instance.dns.guid
  locations {
    subnet_crn = ibm_is_subnet.front.crn
    enabled    = true
  }
  locations {
    subnet_crn = ibm_is_subnet.front.crn
    // fix by using back instead of front
    // subnet_crn = ibm_is_subnet.back.crn
    enabled = true
  }
}
```

**Comment by MalarvizhiK** [comment]
@powellquiring I am able to recreate the issue. We will work on a fix.

**Comment by hkantare** [comment]
Fixed in latest release 1.42.0
